src/components/app/ServiceWorkerManager.js

```diff
@@ -0,0 +1,152 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+
+import React, { PureComponent } from 'react';
+import explicitConnect from '../../utils/connect';
+
+import { getDataSource } from '../../reducers/url-state';
+import { getView } from '../../reducers/app';
+
+import type {
+  ExplicitConnectOptions,
+  ConnectedProps,
+} from '../../utils/connect';
+import type { DataSource } from '../../types/actions';
+import type { Phase } from '../../types/reducers';
+
+import './ServiceWorkerManager.css';
+
+type StateProps = {|
+  +dataSource: DataSource,
+  +phase: Phase,
+|};
+type Props = ConnectedProps<{||}, StateProps, {||}>;
+
+type InstallStatus = 'pending' | 'ready' | 'idle';
+type State = {|
+  installStatus: InstallStatus,
+  closeNotice: boolean,
+|};
+
+class ServiceWorkerManager extends PureComponent<Props, State> {
+  state = {
+    installStatus: 'idle',
```

Comment:
I discussed with my inner self about the location of this state: as a local state here and as a redux state. A redux state would make it possible to use it in other components obviously. First I wrote a redux state but then decided it was too much code for what I needed for now, and used a local state instead.

Reply:
Yeah, I think it's okay to be a state.